README.md

```diff
@@ -79,6 +100,56 @@ Arguments:
 
 <truncated>
 ```
+    
+# Debugging
+Debugging code requires your current `dotnet new3` session to have its active build session configured to DEBUG, and a debug process from Visual Studio to be attached to the current running `dotnet new3` session. The steps required to accomplish this are outlined below.
+
+## Setup
+
+Start by opening up the **Microsoft.Templating.sln** in Visual Studio. This solution contains the projects needed to run, modify & debug the application. Once your solution is loaded, open up a new command prompt and navigate to the root of your repository. At this point, you can take one of two paths.
+
+Path 1:
+- Run the "setup" script that applies to your environment. 
+- Once complete, type "debug" and press enter.
+  - The "debug" command is an alias that kicks off a command to run a script named **dn3buildmode-debug.cmd**. 
+    - Of which sets the build to a DEBUG mode.
+
+Path 2:
+- Execute **dn3buildmode-debug.cmd** prior to running your "setup" script the first time. 
+  - This will save you from having to run "setup" to reach a command prompt that you can run the "debug" command against. 
+
+After your build is set to DEBUG mode, run the "setup" script to build a new `dotnet new3` session. 
+
+## Execution
+
+Once "setup" has completed successfully, run the following command.
+```bash
+dotnet new3 --debug:attach {{additonal args}}
+```
+By supplying the `--debug:attach` argument with any other argument(s) you are running, you are triggering a ` Console.ReadLine();` request which pauses execution of the Template Engine at an early point in its logical loop.
+
+Once the engine is "paused", you have the opportunity to attach a debugger to the running `dotnet new3` process. 
+
+In Visual studio...
+
+- Open the **Microsoft.TemplateEngine.Cli.New3Command** class and locate the following function.
+  - `New3Command.Run()`
+- Set a breakpoint after the following block of code inside the "Run" function.
```

Comment:
The location of the breakpoint doesn't matter, it should be placed in the code the user is interested in rather than immediately after the code that checks for the flag